MagazineLayout/Public/MagazineLayout.swift

```diff
@@ -367,23 +367,22 @@ public final class MagazineLayout: UICollectionViewLayout {
     // See comment in `layoutAttributesForElementsInRect:` for more details.
     guard !hasDataSourceCountInvalidationBeforeReceivingUpdateItems else { return nil }
 
-    let itemLocation = ElementLocation(indexPath: indexPath)
 
-    guard
-      itemLocation.sectionIndex < modelState.numberOfSections(.afterUpdates),
-      itemLocation.elementIndex < modelState.numberOfItems(inSectionAtIndex: itemLocation.sectionIndex, .afterUpdates)
-    else
-    {
-      // On iOS 9, `layoutAttributesForItem(at:)` can be invoked for an index path of a new item
-      // before the layout is notified of this new item (through either `prepare` or
-      // `prepare(forCollectionViewUpdates:)`). This seems to be fixed in iOS 10 and higher.
-      assertionFailure("`{\(itemLocation.sectionIndex), \(itemLocation.elementIndex)}` is out of bounds of the section models / item models array.")
+    // For reasons that are not clear, this function can be invoked with with an out-of-bounds
+    // index path. To guard against downstream index-out-of-bounds crashes, we return nil here if we
+    // detect this scenario.
 
-      // Returning `nil` rather than default/frameless layout attributes causes internal exceptions
-      // within `UICollectionView`, which is why we don't return `nil` here.
```

Comment:
I wasn't able to repro this internal exception, and we return nil in other places, so seems ok to return nil